pipelines/latiss/cpFlat.yaml

```
@@ -21,20 +21,25 @@ tasks:
       doFlat: False
       doFringe: False
       doApplyGains: False
+  cpFlatMeasure:
+    class: lsst.cp.pipe.cpFlatNormTask.CpFlatMeasureTask
+    config:
+      connections.inputExp: 'cpFlatProc'
+      connections.outputStats: 'flatStats'
   cpFlatNorm:
     class: lsst.cp.pipe.cpFlatNormTask.CpFlatNormalizationTask
     config:
-      connections.inputExps: 'isr_metadata'
-      connections.outputScales: 'cpFlatNorm_yaml'
+      connections.inputMDs: 'flatStats'
+      connections.outputScales: 'cpFlatNormScales'
   cpCombine:
-    class: lsst.cp.pipe.cpFunctions.CalibCombineTask
+    class: lsst.cp.pipe.cpCombine.CalibCombineTask
     config:
       connections.inputExps: 'cpFlatProc'
-      connections.inputScales: 'cpFlatNorm_yaml'
+      connections.inputScales: 'cpFlatNormScales'
       connections.outputData: 'flatProposal'
       calibrationType: 'flat'
       calibrationDimensions: ['physical_filter']
       exposureScaling: InputList
-      scalingLevel: Amplifier
+      scalingLevel: AMP
 contracts:
   - isr.doFlat == False
```

In general, what are the "contracts"? (forgive me for the naïve questions). I see that under "isr" you already have "doFlat: False"; why put that here again?

---

The contract is a way to force something to be set one way, with the pipeline construction process failing if the contract is violated (so it fails fast instead of running with the wrong configuration).  Contracts can also be used to configure parameters to match between tasks (like a list of mask planes to be set in one task, and then ignored in a second task).